### PR TITLE
Massively Buffs the Lightning Bolt Spell

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Note that when I'm talking about the "Lightning Bolt" spell in this PR, I'm NOT talking about the spell that summons a "spell packet" that stuns people when thrown (that's the "Thrown Lightning" spell) OR the spell that slowly charges up an electrical blast that can stun people when it's released (that's the "Tesla Blast" spell). I'm INSTEAD talking about the spell that's actually named "Lightning Bolt", which, quoth the wiki, "Charges a lightning ball. Click a target to throw it. The bolt will arc and split damage between people hit. Does not stun. Is not blocked by hardsuits or insulated gloves.".

Yes, having 3 different spells that shoot lightning at people is confusing, but they do all technically fill different niches (and having some variety in your Thor or Emperor Palpatine gimmick is nice).

ANYWAY, on to what this PR actually does. This PR:

- Makes the Lightning Bolt spell robeless.
- Reduces the maximum cooldown time of the Lightning Bolt spell from 20 seconds to 10 seconds.
- Reduces the minimum cooldown time of the Lightning Bolt spell from 3 seconds to ~~1 second~~ 2 seconds.
- Reduces the point cost of the Lightning Bolt spell from 3 points to 1 point.
- Makes the activation message of the Lightning Bolt spell say that you energize your "hands" instead of just your "hand".
- Changes the invocation message of the Lightning Bolt spell from "UN'LTD P'WAH" to "P'WAH, UNLIM'TED P'WAH".
- Changes the name of the spellbook entry for the Thrown Lightning spell from "Lightning bolt! Lightning bolt!" to "Thrown Lighting".

## Why It's Good For The Game

Now, after reading that list of changes, you might be thinking to yourself: "ATHATH, this is absurd! Lightning bolt is a 3 point spell! Making it robeless and making it cost only 1 point would make it totally unbalanced!"

And to that, I respond: Do you know how absolutely, bafflingly TERRIBLE the Lightning Bolt spell currently is?

To purchase this spell, you have to spend THREE wizard points on it. For reference, the only other 3 point spells currently in the game are Lesser Summon Guns and Arcane Barrage (which are pretty darn strong spells), and even those spells are considered to be slightly overpriced. Well, Lightning Bolt must be a pretty strong spell to compete with the likes of LSG and AB, right?

Wrong. Dead wrong. Literally the opposite of right. Lightning Bolt gives you the ability to, once every 20 SECONDS, fire a single projectile that will cause a tesla_zap() to occur where it hits. If it hits someone directly, it'll deal an extra 15 burn damage to them (and only them), but only if they don't have anti-magic (if the projectile directly hits someone who has anti-magic, the spell fizzles, sort of like how bolts of fireball work). This tesla_zap() is absolutely pathetic. It ignores insulative gear, yes, but it does jack shit to silicons. It does not stun in any way, and the damage the tesla_zap() chain deals is absolutely pathetic: 33+/-5 damage to the first living target, 22+/-5 damage to the second living target, 14+/-5 damage to the third living target, 9+/-5 damage to the fourth living target, and so on. This spell won't even cause enough damage to cause slowdown to your non-primary target(s).

The BEST you can hope for with this spell is to deal a whopping 38-58 damage to a target that you directly hit with this spell (with some minor scratch damage spread out across some of the secondary targets) once every 20 seconds (or as "often" as once every 11.5 seconds if you blow 9 of your 10 wizard points on taking this spell 3 times). For reference, the Thrown Lighting spell costs 1 point less than Lightning Bolt, hardstuns whomever you directly hit with it, and had a maximum (and minimum) cooldown of 1 second. Fireball costs 1 point less than Lightning Bolt, is robeless, deals large amounts of damage **and knockdown** to pretty much everyone in its AoE, and has a maximum cooldown of 6 seconds.

I hope it's plain to see here that Lightning Bolt is absolutely pathetic in comparison, not because those other spells are overtuned (when did you last see a wizard take Thrown Lighting?), but because Lightning Bolt is hilariously undertuned.

The changes made by this PR make Lightning Bolt a cheap AoE damage spell that can be an alternative to Fireball for kill-confirming, dealing damage to magic-immune people, and such. Making it cost only 1 point lets you buy it 5 times (giving it a cooldown of just ~~1 second~~ 2 seconds so you can spam it) and still have 5 wizard points to spare (which you can then spend on other lightning-themed spells, for example).

The activation message change was just because the deactivation message says that you let energy flow out of your HANDS, so I figured that activating the spell should "energize" both of your hands, not just one of them.

The invocation change was to make the reference a bit more obvious, since "UN'LTD" looks more like "UNTOLD" than "UNLIMITED" (and the full version of the line being referenced is "POWER, UNLIMITED POWER!", not just "UNLIMITED POWER!").

The spellbook entry name change was made to reduce confusion and because it's probably a good idea to refer to the spell named "Thrown Lighting" as "Thrown Lightning" instead of "Lightning bolt! Lightning bolt!" in its spellbook entry.

Correct me if I'm wrong, but since I'm not editing any functions here (just variables), I don't think I need to do any codedocing in this PR, right?

~~Also, if y'all think that bringing the minimum cooldown down to 1 second might buff Lightning Bolt TOO much, I wouldn't be opposed to making the minimum cooldown 2 seconds (or 1.5 seconds?) instead.~~

Should I change the spellbook entry for Lighting Bolt to mention that a significant portion of its damage comes from directly hitting people with the bolt projectile itself?

## Changelog
:cl: ATHATH
balance: The Lightning Bolt spell is now robeless, has lower cooldowns, and is cheaper.
spellcheck: "Lightning bolt! Lightning bolt!" is now renamed to "Thrown lightning".
/:cl:

cobby edit: cl

### CODE REVIEW

**Comment by Fikou** [comment]
did you have to put the wall of text in the changelog too

**Comment by 81Denton** [comment]
can I have uhhhhhhhhhhhh tl;dr

**Comment by anconfuzedrock** [comment]
spell only do like 30 damage so now it's 1 point instead of 3 and robeless and double fast because it was somehow worse than spellcards

**Comment by ArcaneMusic** [comment]
Well, it sounds like you've brought lightning bolt up to par with lightning bolt's power, which is a good change, but I wouldn't be against lowering the cost to 2 then just so that you can't combine the two spells as effectively without serious investment in the combo.

Still, overall good change (Summarize your changelog a bit)

**Comment by tralezab** [comment]
just think of it as on par with fireball since it played the same and was way worse

**Comment by necromanceranne** [comment]
Just to double check doesn't lightning bolt check for magic robes when determining if the lightning arcs to people?

Also this combined with tesla shock is far from shit. I've considered it mediocre but one of the worst spells it is not. It just has to be paired with tesla. At which point it can kill entire rooms to in two spell casts.

**Comment by ATH1909** [comment]
> Just to double check doesn't lightning bolt check for magic robes when determining if the lightning arcs to people?
> 
> Also this combined with tesla shock is far from shit. I've considered it mediocre but one of the worst spells it is not. It just has to be paired with tesla. At which point it can kill entire rooms to in two spell casts.

I didn't see any code about that, but if someone finds it, please do tell me.

And, uh, you sure it's a room-wiper? The damage from tesla blast's lighting thing appears to go from 30->25->20->15->etc. (based on how many targets the lightning thing has gone through so far). 30+58 (highroll for Lightning Bolt)=88, which shouldn't be enough to kill on its own.

**Comment by LemonInTheDark** [comment]
> 
> I didn't see any code about that, but if someone finds it, please do tell me.
> 
> And, uh, you sure it's a room-wiper? The damage from tesla blast's lighting thing appears to go from 30->25->20->15->etc. (based on how many targets the lightning thing has gone through so far). 30+58 (highroll for Lightning Bolt)=88, which shouldn't be enough to kill on its own.

The tesla damage function looks like `(min(round(power/600), 90) + rand(-5, 5))`, with power dropping by 1/3 each zap if it hits all humans. I don't know the starting numbers.

**Comment by ATH1909** [comment]
> > I didn't see any code about that, but if someone finds it, please do tell me.
> > And, uh, you sure it's a room-wiper? The damage from tesla blast's lighting thing appears to go from 30->25->20->15->etc. (based on how many targets the lightning thing has gone through so far). 30+58 (highroll for Lightning Bolt)=88, which shouldn't be enough to kill on its own.
> 
> The tesla damage function looks like `(min(round(power/600), 90) + rand(-5, 5))`, with power dropping by 1/3 each zap if it hits all humans. I don't know the starting numbers.

You're thinking of the equation for Lightning Bolt. 30->25->20->etc. is the equation for Tesla Blast. C'mon, I reminded you that these were two different spells in the opening post of this PR.

**Comment by ExcessiveUseOfCobblestone** [comment]
you don't have to put every small thing changed in the CL, especially in such a verbose format.

**Comment by AdamElTablawy** [comment]
The reason it sucks so much is because it used to stun.

Now that it doesn't, it's basically useless, but I'm not sure making it a robeless spell is a good idea.

Also being able to shoot this every second and have it ignore insulative gear is going to be quite a meme, frankly. One I want to see, but it's still worth pointing out it might be a little OP.

**Comment by Spepbuthewranglesgithub** [comment]
> Also being able to shoot this every second and have it ignore insulative gear is going to be quite a meme, frankly. One I want to see, but it's still worth pointing out it might be a little OP.

ain't that kind of the point of wizards.

**Comment by AdamElTablawy** [comment]
Yeah but there's a point where OP becomes unfun, which is why ei nath doesn't work on, say, people with holy items.

Fun OP good. But I forsee people getting real sick real fast of a wizard that does 50 dmg every 1 second with proper aim without the risk of blasting himself like with fireball.

**Comment by ExcessiveUseOfCobblestone** [comment]
"wizard is suppose to be op" is how it gets disabled on lowpop and manuel.

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 1]
```diff
@@ -135,7 +135,7 @@
 	category = "Defensive"
 
 /datum/spellbook_entry/lightningPacket
-	name = "Lightning bolt!  Lightning bolt!"
```
Comment (lordpidey) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]: Aww, I thought the joke was funny.

https://www.youtube.com/watch?v=j_ekugPKqFw

**Comment by optimumtact** [comment]
relax your changelog brother

**Comment by ATH1909** [comment]
@optimumtact @ExcessiveUseOfCobblestone Is this changelog better?

**Comment by ExcessiveUseOfCobblestone** [comment]
you should consider making this 2 points, fireball is 1 because it requires you to aim it right or it can kill you faster than anyone else on the station could.

I don't want seeking 30+ damage spam that chains on top of stuns from the other spells. 

I'd personally rather see something undertuned than yet another mass kill device on the wizard on top of the existing ones, the gamemode has no(/extremely little) scaling whatsoever which is why it gets flat out removed from rotation when you start getting things like lower pop.

**Comment by Shadowflame909** [comment]
What cobby said or add a aoe that can self-shock if you're too close.

We need more fireball spells for different elements.

**Comment by ATH1909** [comment]
> you should consider making this 2 points, fireball is 1 because it requires you to aim it right or it can kill you faster than anyone else on the station could.
> 
> I don't want seeking 30+ damage spam that chains on top of stuns from the other spells.
> 
> I'd personally rather see something undertuned than yet another mass kill device on the wizard on top of the existing ones, the gamemode has no(/extremely little) scaling whatsoever which is why it gets flat out removed from rotation when you start getting things like lower pop.

What if I made the minimum cooldown 2 seconds instead of 1 second?

**Comment by ExcessiveUseOfCobblestone** [comment]
sorry I don't play ~~wizard~~, that's even more reason to make it 2.

**Comment by ATH1909** [comment]
> sorry I don't play ~wizard~, that's even more reason to make it 2.

Fireball costs 2 points because you can knock people down/out with it and its maximum cooldown time is 6 seconds instead of 10 seconds.

> I don't want seeking 30+ damage spam that chains on top of stuns from the other spells.

This spell is _meant_ to synergize with stun spells; this is a way for wizards to confirm their kills in a stylish, Palpatine-esque manner. You only get good amounts of damage from this spell if you get direct hits with it; the chain damage is pretty awful.

If you'd like, I can increase the cooldown time of this spell or make it deal less damage (on either the chain lightning or on direct hits), but I'm not budging on its point cost.

**Comment by ExcessiveUseOfCobblestone** [comment]
I tested this and am not satisfied with a 1 second spell that gives you chained targeting and crits in 3 hits on top of 5 remaining points (I put 4 in Tesla so it was just stun then direct > direct > direct and now 1 person crit, 1 person red, 1 person orange/yellow, and I could have easily did 5/6 hits).

If you are not going to budge on the point cost I recommend changing it in other aspects so it acts more as a followup to say tesla than a rapid fire device. It's too good right now.

**Comment by senjx** [comment]
stop changing things

**Comment by ATH1909** [comment]
> I tested this and am not satisfied with a 1 second spell that gives you chained targeting and crits in 3 hits on top of 5 remaining points (I put 4 in Tesla so it was just stun then direct > direct > direct and now 1 person crit, 1 person red, 1 person orange/yellow, and I could have easily did 5/6 hits).
> 
> If you are not going to budge on the point cost I recommend changing it in other aspects so it acts more as a followup to say tesla than a rapid fire device. It's too good right now.

Did you test it with a 2 second delay?

**Comment by ExcessiveUseOfCobblestone** [comment]
no i just tested what you PRd

**Comment by Iamgoofball** [comment]
>Changes the name of the spellbook entry for the Thrown Lightning spell from "Lightning bolt! Lightning bolt!" to "Thrown Lighting".

bad change

**Comment by ATH1909** [comment]
Alright, I reduced the minimum cooldown time of this spell to 2 seconds.

I kind of feel like it's lost a bit of the "shoot lightning continuously at everything" theme that I was going for, though. Should I change it back to 1 second and just reduce the amount of damage you deal on a direct hit (or indirect hit?) or something (and if so, by how much?)?

**Comment by ATH1909** [comment]
> > Changes the name of the spellbook entry for the Thrown Lightning spell from "Lightning bolt! Lightning bolt!" to "Thrown Lighting".
> 
> bad change

Casting the spell (or rather, throwing the spellpackets from that spell) still references that meme, y'know. We don't have to sacrifice clarity in and easy usage of the spellbook in order to reference memes, we can do that in the invocations and special interactoins of spells.

**Comment by ExcessiveUseOfCobblestone** [comment]
we can try this and I can be mad about it later.

**Pull request merged**